src/Integration.Vsix/CFamily/VcxProject/FileConfig.cs

```diff
@@ -37,76 +35,45 @@ public static FileConfig TryGet(ILogger logger, ProjectItem dteProjectItem, stri
             }
 
             var vcConfig = vcProject.ActiveConfiguration;
-            var platformName = ((VCPlatform)vcConfig.Platform).Name; // "Win32" or "x64"
-            var includeDirs = vcConfig.GetEvaluatedPropertyValue("IncludePath");
-            var platformToolset = vcConfig.GetEvaluatedPropertyValue("PlatformToolset");
-            var vcToolsVersion = vcConfig.GetEvaluatedPropertyValue("VCToolsVersion");
-            var compilerVersion = GetCompilerVersion(platformToolset, vcToolsVersion);
 
             var vcFileSettings = GetVcFileSettings(logger, absoluteFilePath, vcConfig, vcFile);
-
             if (vcFileSettings == null)
             {
                 // Not supported
                 return null;
             }
+            CmdBuilder cmdBuilder = new CmdBuilder(vcFile.ItemType == "ClInclude");
 
-            // Fetch properties that can be set differently for header files
-            var compileAs = vcFileSettings.GetEvaluatedPropertyValue("CompileAs");
-            var forcedIncludeFiles = vcFileSettings.GetEvaluatedPropertyValue("ForcedIncludeFiles");
-            var precompiledHeader = vcFileSettings.GetEvaluatedPropertyValue("PrecompiledHeader");
-            var precompiledHeaderFile = vcFileSettings.GetEvaluatedPropertyValue("PrecompiledHeaderFile");
-            
-            if (vcFile.ItemType == "ClInclude")
+            var compilerPath = vcConfig.GetEvaluatedPropertyValue("ClCompilerPath");
+            if (string.IsNullOrEmpty(compilerPath))
             {
-                // If the project language is not specified. Headers are compiled as CPP
-                compileAs = (compileAs == "Default") ? "CompileAsCpp" : compileAs;
-                // Use PCH in header files if project PCH is configured and not enforced through force include
-                // In normal code, it should be self to assume this since PCH should be used on every CPP file
-                if (string.IsNullOrEmpty(forcedIncludeFiles) && precompiledHeader == "Use")
+                // in case ClCompilerPath is not available on VS2017
```

Comment:
@abbas-sabra-sonarsource we've dropped support for VS 2017, so this code would only run in VS 2019+. Does this have any implications on the code below? Can it be simplified?

Reply:
Even though we dropped support to VS 2017, its toolchain can still be used with newer vs. So better to keep it. maybe we can improve the comment to say `VS2017 toolchains`

Reply:
I'll update the comment, toolchains helps to understand the problem much better 👍